File: process_logo.py
from PIL import Image
import sys

def make_transparent(img_path):
    try:
        img = Image.open(img_path)
        img = img.convert("RGBA")
        datas = img.getdata()

        newData = []
        for item in datas:
            r, g, b, a = item
            # The grey background is usually a light color where R, G, B are very close to each other.
            # Zameen Gem logo colors are Gold (R: ~212, G: ~175, B: ~55) and Dark Green.
            if abs(r - g) < 25 and abs(r - b) < 25 and abs(g - b) < 25 and r > 180 and g > 180 and b > 180:
                newData.append((255, 255, 255, 0)) # transparent
            else:
                newData.append(item)

        img.putdata(newData)
        img.save(img_path, "PNG")
        print("Successfully made the background transparent.")
    except Exception as e:
        print(f"Error: {e}")
        sys.exit(1)

make_transparent("./public/images/zameen_gem_logo.png")

File: process_logo.js
const Jimp = require('jimp');

async function removeGrey() {
  const image = await Jimp.read('./public/images/zameen_gem_logo.png');
  
  // To avoid jagged edges, we'll implement a simple threshold transparency
  // where greys/whites become transparent.
  
  image.scan(0, 0, image.bitmap.width, image.bitmap.height, function(x, y, idx) {
    const r = this.bitmap.data[idx + 0];
    const g = this.bitmap.data[idx + 1];
    const b = this.bitmap.data[idx + 2];
    
    const max = Math.max(r, g, b);
    const min = Math.min(r, g, b);
    const diff = max - min;
    
    // Grey/White usually has a low difference between RGB channels.
    // If it's grey/white, make it transparent. 
    // Assuming the logo is Gold/Royal Blue. Gold has high diff (R=212, G=175, B=55).
    // Royal Blue has high diff.
    if (diff < 25 && max > 120) {
      this.bitmap.data[idx + 3] = 0; // make transparent
    }
  });

  await image.writeAsync('./public/images/zameen_gem_logo.png');
  console.log("Processed logo successfully");
}

removeGrey().catch(err => {
  console.error("Error processing logo:", err);
  process.exit(1);
});
